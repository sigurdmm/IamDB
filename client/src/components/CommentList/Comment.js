import PropTypes from 'prop-types';

import { dateToHuman } from '../../utils/dateUtil';
import './Comment.less';

const Comment = ({ createdAt, children }) => <div className="comment">
    <strong className="comment__published">
      Published: <em>{dateToHuman(new Date(parseInt(createdAt, 10)))}</em>
    </strong>
    <p className="comment__text">{children}</p>
  </div>;
Comment.propTypes = {
  createdAt: PropTypes.any,
  children: PropTypes.string,
};

export default Comment;
