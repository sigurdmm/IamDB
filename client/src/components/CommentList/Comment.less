@import '../../constants';

.comment {
  background-color: @white;
  box-shadow: rgba(0,0,0,.05) 1px 2px 4px;
  box-sizing: border-box;

  padding: 0.5em;
  border-top: @primaryColor solid 1px;

  &__published {
    color: @primaryColor;
  }

  &__text {
    font-size: 0.8em;
  }
}
