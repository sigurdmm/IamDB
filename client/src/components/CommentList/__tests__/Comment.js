import { render } from 'enzyme';

import Comment from '../Comment';

describe('<Comment/>', () => {
  it('should render properly', () => {
    const wrapper = render(<Comment createdAt={'1542296465964'}>Hello world</Comment>);

    expect(wrapper).toMatchSnapshot();
  });
});
