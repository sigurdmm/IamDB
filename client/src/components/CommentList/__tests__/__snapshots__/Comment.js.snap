// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment/> should render properly 1`] = `
<div
  class="comment"
>
  <strong
    class="comment__published"
  >
    Published: 
    <em>
      15/10/2018
    </em>
  </strong>
  <p
    class="comment__text"
  >
    Hello world
  </p>
</div>
`;
